src/libServer/LookupServer.cpp

```diff
@@ -876,7 +881,58 @@ Json::Value LookupServer::GetBalance(const string& address) {
     throw JsonRpcException(RPC_MISC_ERROR, "Unable To Process");
   }
 }
+string LookupServer::GetEthCall(const Json::Value& _json) {
+  LOG_MARKER();
+  const auto& addr = JSONConversion::checkJsonGetEthCall(_json);
+  // Add mutex
+  uint64_t gasRemained =
+      2 * DS_MICROBLOCK_GAS_LIMIT;  // for now set total gas as twice the ds gas
+                                    // limit
+  bytes code{};
+  bool ret = false;
+  {
+    shared_lock<shared_timed_mutex> lock(
+        AccountStore::GetInstance().GetPrimaryMutex());
+    Account* contractAccount =
+        AccountStore::GetInstance().GetAccount(addr, true);
+    if (contractAccount == nullptr) {
+      throw JsonRpcException(RPC_INVALID_PARAMS, "Account does not exist");
+    }
+    code = contractAccount->GetCode();
+  }
+  Address fromAddr;
+  string result;
+  uint64_t amount{0};
+  try {
+    if (_json.isMember("fromAddr")) {
```

Comment:
fromAddr makes sense to me, but for some reason CreateTransaction uses pubkey, and derives the address from it.  I wonder whether we should do the same here or not.

Reply:
I used the eth_call config for this purpose however happy to change it

Reply:
I am not sure which way is better, just thought that mimicking `CreateTransaction` would make more sense for consistency of the Zilliqa API